src/GCRealTimeMon/Configuration/NewConfigurationManager.cs

```diff
@@ -0,0 +1,138 @@
+            if (shouldSetupTimer)
+            {
+                // Try to grab the defaults from the specified default config.
+                // Since the stats_mode is optional, check for the existence of the timer.
+                string defaultTimer = null;
+                if (defaultConfiguration != null &&
+                    defaultConfiguration.StatsMode != null && 
+                    defaultConfiguration.StatsMode.TryGetValue("timer", out string timerValue))
+                {
+                    defaultTimer = timerValue; 
+                }
+
+                string timer = Prompt.Input<string>(message: "Enter the period magnitude as an integer and 'm' for minutes / 's' for seconds",
+                                                    defaultValue: defaultTimer);
+                statsHeapDict["timer"] = timer;
+            }
+
+            // Min GC Pause In Msec.
+            Console.WriteLine();
+            bool shouldSetupMinGCPauseInMsec = Prompt.Confirm("Would you like to set a value for the minimum GC Pause duration to filter GCs off of?");
+            Dictionary<string, string> displayConditions = defaultConfiguration?.DisplayConditions ?? new();
+            if (shouldSetupMinGCPauseInMsec)
+            {
+                string defaultGCPauseDuration = null;
+                if (defaultConfiguration != null &&
+                    defaultConfiguration.DisplayConditions != null &&
+                    defaultConfiguration.DisplayConditions.TryGetValue("min gc duration (msec)", out string minDuration))
+                {
+                    defaultGCPauseDuration = minDuration;
+                }
+
+                string minGCPauseTime = Prompt.Input<string>(message: "Enter the minimum GC Pause duration value to consider in Msec",
+                                                             defaultValue: defaultGCPauseDuration);
+                displayConditions["min gc duration (msec)"] = minGCPauseTime;
+            }
+
+            Configuration configuration = new Configuration
+            {
+                Columns = nameOfColumns.ToList(),
+                AvailableColumns = Utilities.ColumnInfoMap.Map.Keys.ToList()
+            };
+
+            if (shouldSetupTimer)
+            {
+                configuration.StatsMode = statsHeapDict;
+            }
+            if (shouldSetupMinGCPauseInMsec)
+            {
+                configuration.DisplayConditions = displayConditions;
+            }
+
+            // Validate the configuration before persisting.
+            ConfigurationReader.ValidateConfiguration(configuration);
+
+            var serializer = new SerializerBuilder()
+                                 .WithNamingConvention(UnderscoredNamingConvention.Instance)
+                                 .Build();
+            var serializedResult = serializer.Serialize(configuration);
+            await File.WriteAllTextAsync(path: path, contents: serializedResult);
```

Comment:
Is there any chance that path could be null or invalid?
Maybe at least protect against the invalid situation with a try/catch and transform into:
`   throw new ArgumentException($"Invalid configuration file path '{path}'");`